engine.py

```diff
@@ -912,3 +912,126 @@ def __setup_favorite_dirs(self):
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self, session=None):
+        """
+        Returns the absolute path to the current file if it resides
+        on disk. If the session has never been saved and isn't
+        associated with a file on disk yet, an empty string is returned.
+
+        :param session: An object representing the active document
+                        (for MDI applications).
+        :returns: Path to the current file if it has been saved to disk,
+                  else returns an empty string.
+        :raises TankError: Raises a `TankError` if the application is
+                           unable to determine the session that is
+                           being referred to.
+        """
+        if self.hiero_enabled or self.studio_enabled:
+            import hiero
+
+            # first find what the current project is. Hiero is a multi
+            # project environment so we can ask the engine which
+            # project was last clicked.
+            selection = self.get_menu_selection()
+
+            # to guard against multiple selections, make sure we have
+            # only one selection.
+            if len(selection) != 1:
+                raise tank.TankError("Please select a single project.")
+            if not isinstance(selection[0], hiero.core.Bin):
+                raise tank.TankError("Please select a Hiero project.")
+
+            project = selection[0].project()
+
+            if project is None:
+                # apparently bins can be without projects (child bins)
+                raise tank.TankError("Please select a Hiero Project.")
+
+            session_path = project.path()
+        else:
+            session_path = nuke.root().knob("name").value()
+
+        # replace forward slashes with the OS-specific path separator
+        # to make Nuke happy on Windows.
+        return session_path.replace("/", os.path.sep)
+
+    def get_session_dependencies(self, session=None):
+        """
+        Returns a list of file dependencies for the current session.
+
+        :param session: An object representing the active document
+                        (for MDI applications).
+        :returns: A list of file dependencies required to load
+                  the session. The data returned is of the form:
+                  [
+                    {"path": "/foo/bar/hello.%04d.jpeg",
+                     "engine": "tk-nuke",
+                     "type": "Read"
+                     },
+                    {"path": "/foo/bar/hello.%04d.obj",
+                     "engine": "tk-nuke",
+                     "type": "ReadGeo2"
+                     },
+                  ]
+         """
+        dependencies = []
+
+        if self.hiero_enabled or self.studio_enabled:
+            import hiero.core
+
+            # collect dependencies for all projects opened
```

Comment:
i think it's possibly asymmetrical to get all the dependencies here where you only get the file name for one project when you call `get_session_path()` - i think we should apply the same logic as we do above and try to find out what the current session is and then get deps for it.